### PR TITLE
[GLUTEN-7004][CORE] Bump Spark version to 3.4.3

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Spark 3.4.3 is the latest release version of the 3.4 branch.

(Fixes: \#7004)

### CODE REVIEW

**Comment by Yohahaha** [comment]
```
/tmp/jnilib-1835571804442270261.tmp(+0x1299c68)[0x7fb8e4ca2c68]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow4util8ArrowLogD1Ev+0xed)[0x7fb8e4ca2f3d]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow7dataset3jni31ReservationListenableMemoryPool4FreeEPhll+0x461)[0x7fb8e4291be1]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow10PoolBufferD0Ev+0x47)[0x7fb8e4f345e7]
/tmp/jnilib-1835571804442270261.tmp(_ZNSt6vectorISt10shared_ptrIN5arrow6BufferEESaIS3_EED1Ev+0x6b)[0x7fb8e43a378b]
/tmp/jnilib-1835571804442270261.tmp(_ZNSt23_Sp_counted_ptr_inplaceIN5arrow9ArrayDataESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv+0xda)[0x7fb8e44af68a]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow17SimpleRecordBatchD1Ev+0x116)[0x7fb8e4ddc006]
/tmp/jnilib-1835571804442270261.tmp(_ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE10_M_releaseEv+0x48)[0x7fb8e428ab08]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt10shared_ptrINS_11RecordBatchEEE20WrapResultOnComplete8CallbackINSC_14ThenOnCompleteIZNS_23DefaultIfEmptyGeneratorISB_EclEvEUt_NSC_17PassthruOnFailureISI_EEEEEEED0Ev+0x2e)[0x7fb8e42ddc7e]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x3a4)[0x7fb8e4cbbbf4]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow6FutureISt10shared_ptrINS_11RecordBatchEEE12MarkFinishedENS_6ResultIS3_EE+0xda)[0x7fb8e42fbdda]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt10shared_ptrINS_11RecordBatchEEE20WrapResultOnComplete8CallbackINS_6detail16MarkNextFinishedISC_SC_Lb0ELb0EEEEEE6invokeES4_+0x56)[0x7fb8e42fc9b6]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl21RunOrScheduleCallbackERKSt10shared_ptrINS_10FutureImplEEONS2_14CallbackRecordEb+0x46)[0x7fb8e4cbc0c6]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow10FutureImpl11AddCallbackENS_8internal6FnOnceIFvRKS0_EEENS_15CallbackOptionsE+0xcb)[0x7fb8e4caa95b]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt8functionIFNS8_ISt10shared_ptrINS_11RecordBatchEEEEvEEE20WrapResultOnComplete8CallbackINSG_14ThenOnCompleteIZNS_20FutureFirstGeneratorISC_EclEvEUlRKSF_E_NSG_17PassthruOnFailureISO_EEEEEEE6invokeES4_+0xde)[0x7fb8e42fcdae]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(+0x8b679b)[0x7fb8e42bf79b]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(+0x8a79b6)[0x7fb8e42b09b6]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt10shared_ptrINS_3csv15StreamingReaderEEE20WrapResultOnComplete8CallbackINS_6detail16MarkNextFinishedISD_SD_Lb0ELb0EEEEEE6invokeES4_+0x164)[0x7fb8e42dfee4]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(+0x121c8cd)[0x7fb8e4c258cd]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow6FutureINS_8internal5EmptyEE12MarkFinishedIS2_vEEvNS_6StatusE+0xc0)[0x7fb8e42dcaa0]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureINS0_5EmptyEE21WrapStatusyOnComplete8CallbackINS_6detail16MarkNextFinishedISA_SA_Lb1ELb1EEEEEE6invokeES4_+0x28)[0x7fb8e42dcbe8]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl21RunOrScheduleCallbackERKSt10shared_ptrINS_10FutureImplEEONS2_14CallbackRecordEb+0x46)[0x7fb8e4cbc0c6]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow10FutureImpl11AddCallbackENS_8internal6FnOnceIFvRKS0_EEENS_15CallbackOptionsE+0xcb)[0x7fb8e4caa95b]
/tmp/jnilib-1835571804442270261.tmp(_ZNK5arrow6FutureINS_8internal5EmptyEE11AddCallbackINS_6detail16MarkNextFinishedIS3_S3_Lb1ELb1EEENS3_21WrapStatusyOnComplete8CallbackIS7_EEEEvT_NS_15CallbackOptionsE+0x5b)[0x7fb8e4c6951b]
/tmp/jnilib-1835571804442270261.tmp(+0x1249823)[0x7fb8e4c52823]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow6FutureISt10shared_ptrINS_6BufferEEE12MarkFinishedENS_6ResultIS3_EE+0xda)[0x7fb8e466ca2a]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt10shared_ptrINS_6BufferEEE20WrapResultOnComplete8CallbackINS_6detail16MarkNextFinishedISC_SC_Lb0ELb0EEEEEE6invokeES4_+0x56)[0x7fb8e466d636]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl21RunOrScheduleCallbackERKSt10shared_ptrINS_10FutureImplEEONS2_14CallbackRecordEb+0x46)[0x7fb8e4cbc0c6]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow10FutureImpl11AddCallbackENS_8internal6FnOnceIFvRKS0_EEENS_15CallbackOptionsE+0xcb)[0x7fb8e4caa95b]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvRKNS_10FutureImplEEE6FnImplINS_6FutureISt10shared_ptrINS_6BufferEEE20WrapResultOnComplete8CallbackINSC_14ThenOnCompleteIZNS_21TransformingGeneratorISB_SB_E26TransformingGeneratorStateclEvEUlRKSB_E_NSC_17PassthruOnFailureISL_EEEEEEE6invokeES4_+0x99)[0x7fb8e4c67f89]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow18ConcreteFutureImpl22DoMarkFinishedOrFailedENS_11FutureStateE+0x397)[0x7fb8e4cbbbe7]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow6FutureISt10shared_ptrINS_6BufferEEE12MarkFinishedENS_6ResultIS3_EE+0xda)[0x7fb8e466ca2a]
/tmp/jnilib-1835571804442270261.tmp(_ZN5arrow8internal6FnOnceIFvvEE6FnImplIZZNS0_8Executor10DoTransferISt10shared_ptrINS_6BufferEENS_6FutureIS9_EENS_6ResultIS9_EEEENSA_IT_EESF_bENUlRKSD_E0_clESH_EUlvE_E6invokeEv+0x52)[0x7fb8e466cb62]
/tmp/jnilib-1835571804442270261.tmp(+0x129224d)[0x7fb8e4c9b24d]
/tmp/jnilib-1835571804442270261.tmp(+0x1de93e0)[0x7fb8e57f23e0]
/lib64/libpthread.so.0(+0x81ca)[0x7fb9bd1191ca]
/lib64/libc.so.6(clone+0x43)[0x7fb9bc9578d3]
/opt/gluten/ep/_ep/arrow_ep/java/dataset/src/main/cpp/jni_util.cc:
```

only happens in Spark35, not related with current PR.

**Comment by Yohahaha** [comment]
@PHILO-HE @zhouyuan please take a look, thank you!

File: gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -717,6 +718,7 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude(
       "SPARK-26677: negated null-safe equality comparison should not filter matched row groups")
   enableSuite[GlutenParquetV2QuerySuite]
+    .exclude("row group skipping doesn't overflow when reading into larger type")
```
Comment (PHILO-HE) [inline gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:721]: Do you know why this test fails? Please file one issue for it if necessary.

**Review by PHILO-HE — COMMENTED** [review]
@zzcclp, please also take a review.

File: gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala (hunk 1 above)
Comment (Yohahaha) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:721]: Gluten with latest Spark 3.3.x and 3.5.x all disable it, I will open an issue, but I dont know the reason.

Comment (Yohahaha) [reply gluten-ut/spark34/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:721]: #7128

**Review by PHILO-HE — APPROVED** [review]
Looks nice!
@baibaichen, @zzcclp, do you have any comment?

**Comment by baibaichen** [comment]
CH Backend doesn't support 3.4

**Pull request merged**